code/datums/components/onwear_mood.dm

```diff
@@ -0,0 +1,50 @@
+// Add to clothing to give the wearer a mood buff and a unique examine str
+
+/datum/component/onwear_mood
+	/// the event the wearer experiences
+	var/datum/mood_event/saved_event
+	/// examine string added to examine
+	var/examine_string
+
+/datum/component/onwear_mood/Initialize(clear_after, datum/mood_event/saved_event, examine_string)
+
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	if(istype(saved_event))
+		src.saved_event = saved_event
+
+	src.examine_string = examine_string
+
+	if(isnum(clear_after))
```

Comment:
```suggestion
	if(!isnull(clear_after))
```

We want to be able to error if this is something else.

Reply:
wait couldnt you just do if(var) to check null

Reply:
That also catches for FALSE and 0, which you would not want here. People use `if (x)` and `if (!x)` as replacements for isnull but they are not the same, generally I believe you should prefer `isnull` when that is what you want specifically so that you do not hide errors, but we don't enforce it. We would, however, want to be more specific than "isnum" here, otherwise if I pass in a string here I will get no error. That being said, this is part of code I don't think you should have anyway.